rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,499 @@
+    - (+) It would be most intuitive if its indentation logic implements Nix's magic multiline strings.
+- (-) Takes up more vertical space
+- (+) Takes up less horizontal space. (In contrast to, e.g., `##` lines do not have to be prefixed)
+- (+) Only one character to change the semantics of the whole comment. 
+    - (+) Allows to add and remove things from the documentation quickly.
+    - (-) Accidentally adding/removing could happen.
+    
+</details>
+
+
+## CommonMark as the content of doc-comments
+
+**Observing**: The use of CommonMark, a widely recognized and standardized format for documents, is prevalent in the documentation of code and software libraries.
+
+**Considering**: Doc-comments' content should be intuitive to read and write and straightforward to render. Furthermore it should follow established conventions in the nix ecosystem.
+
+**Decision**: CommonMark is the content of all doc-comments.
+
+> Markdown is the most accepted and understood format for writing documentation in tech projects. Also, following the existing RFC-72 is highly profitable and consistent for the Nix ecosystem.
+
+<details>
+<summary>Arguments</summary>
+
+- (+) CommonMark is the official format for Nix documentation; Decided in [RFC72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).
+    - (+) It Would be consistent if this RFC builds upon the previous one.
+    - (+) Further Arguments for CommonMark, in general, can be found in RFC72
+- (+) Allows copy-paste from and to markdown files. We allow easy refactoring if documentation arises and needs to be split into separate files.
+- (-) Strictly binding doc comments content to commonMark might restrict users.
+    - (+) Users/Tools can still use regular comments or develop alternative solutions.
+- (-) CommonMark does not specify the current rich features from nixdoc, such as predefined sections and structures that could be used as a source for automated toolings. Such as types for type-checking or examples to run automated-tests
+    - (+) Future tools can still build their conventions on top of this RFC. They might not directly specify them in an RFC but be a tool developers choose for a specific codebase. However, we have yet to get that tools. So it is good when rich features remain unspecified.
+  
+</details>
+
+## Single-line doc-comments (do not exist)
+
+**Observing**: Nix offers two variants of comments; single- (`#`) and multi-line comments (`/* */`). There may be use cases where it is desirable to have a form of single-line comments subtyped for doc-comment purposes.
+
+**Considering**: Single-line comment for documentation. (Starting a doc-comment with `##`)
+
+**Decision**: Single-line comments (starting with `##`) **cannot be used** in any form for documentation puposes.
+
+<details>
+<summary>Arguments</summary>
+
+- (+) It Would be consistent with providing variants for both nix comments.
+- (-) Doc-comments should have only one variant to reduce complexity.  
```

Comment:
These (+)/(-) seem flipped around, I'd expect (+) to be _for_ the decision and (-) to be _against_ the decision, which doesn't seem to be the case.

Reply:
Changed. +/- is now relative to the 'decision'